call-home/src/bin/callhome/main.rs

```diff
@@ -256,5 +294,81 @@ async fn generate_report(
     report.storage_media = StorageMedia::new(valid_disks);
 
     report.nexus = Nexus::new(event_data);
+
+    let logs = logs.lock().unwrap();
+    for log in logs.iter() {
+        let log_string = format!(
+            "[{} {} {}:{}] {}",
+            log.timestamp, log.level, log.file, log.line, log.message
+        );
+        report.logs.push(log_string);
+    }
     report
 }
+
+// Define the LogsLayer
+struct LogsLayer {
```

Comment:
This bit and below definitely should move up to the lib level, than be at this bin level.